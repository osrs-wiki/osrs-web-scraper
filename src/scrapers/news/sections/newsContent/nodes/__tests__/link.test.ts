import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import linkParser from "../link";

describe("link node", () => {
  test("Links should parse and render", () => {
    const root = parse("<a href='link'>test</a>");
    const builder = new MediaWikiBuilder();
    builder.addContents([linkParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
