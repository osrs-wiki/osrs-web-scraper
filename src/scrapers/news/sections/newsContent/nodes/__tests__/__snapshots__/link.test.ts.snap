// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link node Links should parse and render 1`] = `"[link test]"`;
